File: Nx3CS/bin/test/src/nx3/NPart.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class NPart : global::haxe.lang.HxObject {
		public    NPart(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    NPart(global::Array<object> voices, global::nx3.EPartType type, global::nx3.EClef clef, global::nx3.EDisplayALN clefDisplay, global::nx3.EKey key, global::nx3.EDisplayALN keyDisplay){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NPart.hx"
				global::nx3.NPart.__hx_ctor_nx3_NPart(this, voices, type, clef, clefDisplay, key, keyDisplay);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_NPart(global::nx3.NPart __temp_me71, global::Array<object> voices, global::nx3.EPartType type, global::nx3.EClef clef, global::nx3.EDisplayALN clefDisplay, global::nx3.EKey key, global::nx3.EDisplayALN keyDisplay){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\NPart.hx"
				__temp_me71.nvoices = voices;
				{
					#line 14 "F:\\nx3\\src\\nx3\\NPart.hx"
					int _g = 0;
					#line 14 "F:\\nx3\\src\\nx3\\NPart.hx"
					while (( _g < voices.length )){
						#line 14 "F:\\nx3\\src\\nx3\\NPart.hx"
						global::nx3.NVoice voice = ((global::nx3.NVoice) (voices[_g]) );
						#line 14 "F:\\nx3\\src\\nx3\\NPart.hx"
						 ++ _g;
						#line 14 "F:\\nx3\\src\\nx3\\NPart.hx"
						voice.npart = __temp_me71;
					}
					
				}
				
				#line 16 "F:\\nx3\\src\\nx3\\NPart.hx"
				if (( __temp_me71.nvoices.length > 2 )) {
					#line 18 "F:\\nx3\\src\\nx3\\NPart.hx"
					throw global::haxe.lang.HaxeException.wrap("For now, NPart can\'t have more than two voices");
				}
				
				#line 20 "F:\\nx3\\src\\nx3\\NPart.hx"
				if (( type == default(global::nx3.EPartType) )) {
					#line 20 "F:\\nx3\\src\\nx3\\NPart.hx"
					__temp_me71.type = global::nx3.EPartType.Normal;
				}
				 else {
					#line 20 "F:\\nx3\\src\\nx3\\NPart.hx"
					__temp_me71.type = type;
				}
				
				#line 22 "F:\\nx3\\src\\nx3\\NPart.hx"
				__temp_me71.clef = clef;
				if (( clefDisplay == default(global::nx3.EDisplayALN) )) {
					#line 23 "F:\\nx3\\src\\nx3\\NPart.hx"
					__temp_me71.clefDisplay = global::nx3.EDisplayALN.Layout;
				}
				 else {
					#line 23 "F:\\nx3\\src\\nx3\\NPart.hx"
					__temp_me71.clefDisplay = clefDisplay;
				}
				
				#line 25 "F:\\nx3\\src\\nx3\\NPart.hx"
				__temp_me71.key = key;
				if (( keyDisplay == default(global::nx3.EDisplayALN) )) {
					#line 26 "F:\\nx3\\src\\nx3\\NPart.hx"
					__temp_me71.keyDisplay = global::nx3.EDisplayALN.Layout;
				}
				 else {
					#line 26 "F:\\nx3\\src\\nx3\\NPart.hx"
					__temp_me71.keyDisplay = keyDisplay;
				}
				
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				return new global::nx3.NPart(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				return new global::nx3.NPart(((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (arr[0]) ))) ), ((global::nx3.EPartType) (arr[1]) ), ((global::nx3.EClef) (arr[2]) ), ((global::nx3.EDisplayALN) (arr[3]) ), ((global::nx3.EKey) (arr[4]) ), ((global::nx3.EDisplayALN) (arr[5]) ));
			}
			#line default
		}
		
		
		public virtual   object iterator(){
			unchecked {
				#line 29 "F:\\nx3\\src\\nx3\\NPart.hx"
				return ((object) (new global::_Array.ArrayIterator<object>(((global::Array<object>) (this.nvoices) ))) );
			}
			#line default
		}
		
		
		public  int length;
		
		public virtual   int get_length(){
			unchecked {
				#line 31 "F:\\nx3\\src\\nx3\\NPart.hx"
				return this.nvoices.length;
			}
			#line default
		}
		
		
		public  global::nx3.NBar nbar;
		
		public  global::nx3.EPartType type;
		
		public  global::Array<object> nvoices;
		
		public  global::nx3.EClef clef;
		
		public  global::nx3.EDisplayALN clefDisplay;
		
		public  global::nx3.EKey key;
		
		public  global::nx3.EDisplayALN keyDisplay;
		
		public virtual   global::nx3.NVoice getNVoice(int idx){
			unchecked {
				#line 42 "F:\\nx3\\src\\nx3\\NPart.hx"
				if (( ( idx < 0 ) || ( idx > this.nvoices.length ) )) {
					#line 42 "F:\\nx3\\src\\nx3\\NPart.hx"
					return default(global::nx3.NVoice);
				}
				 else {
					#line 42 "F:\\nx3\\src\\nx3\\NPart.hx"
					return ((global::nx3.NVoice) (this.nvoices[idx]) );
				}
				
			}
			#line default
		}
		
		
		public override   double __hx_setField_f(string field, int hash, double @value, bool handleProperties){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				switch (hash){
					case 520590566:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						this.length = ((int) (@value) );
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return @value;
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return base.__hx_setField_f(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				switch (hash){
					case 701614691:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						this.keyDisplay = ((global::nx3.EDisplayALN) (@value) );
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return @value;
					}
					
					
					case 5343647:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						this.key = ((global::nx3.EKey) (@value) );
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return @value;
					}
					
					
					case 23151640:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						this.clefDisplay = ((global::nx3.EDisplayALN) (@value) );
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return @value;
					}
					
					
					case 1103260490:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						this.clef = ((global::nx3.EClef) (@value) );
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return @value;
					}
					
					
					case 998432111:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						this.nvoices = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return @value;
					}
					
					
					case 1292432058:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						this.type = ((global::nx3.EPartType) (@value) );
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return @value;
					}
					
					
					case 1224747557:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						this.nbar = ((global::nx3.NBar) (@value) );
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return @value;
					}
					
					
					case 520590566:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						this.length = ((int) (global::haxe.lang.Runtime.toInt(@value)) );
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return @value;
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				switch (hash){
					case 1170001818:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getNVoice") ), ((int) (1170001818) ))) );
					}
					
					
					case 701614691:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return this.keyDisplay;
					}
					
					
					case 5343647:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return this.key;
					}
					
					
					case 23151640:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return this.clefDisplay;
					}
					
					
					case 1103260490:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return this.clef;
					}
					
					
					case 998432111:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return this.nvoices;
					}
					
					
					case 1292432058:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return this.type;
					}
					
					
					case 1224747557:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return this.nbar;
					}
					
					
					case 261031087:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("get_length") ), ((int) (261031087) ))) );
					}
					
					
					case 520590566:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						if (handleProperties) {
							#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
							return this.get_length();
						}
						 else {
							#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
							return this.length;
						}
						
					}
					
					
					case 328878574:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("iterator") ), ((int) (328878574) ))) );
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   double __hx_getField_f(string field, int hash, bool throwErrors, bool handleProperties){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				switch (hash){
					case 520590566:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						if (handleProperties) {
							#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
							return ((double) (this.get_length()) );
						}
						 else {
							#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
							return ((double) (this.length) );
						}
						
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return base.__hx_getField_f(field, hash, throwErrors, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				switch (hash){
					case 1170001818:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return this.getNVoice(((int) (global::haxe.lang.Runtime.toInt(dynargs[0])) ));
					}
					
					
					case 261031087:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return this.get_length();
					}
					
					
					case 328878574:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return this.iterator();
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				baseArr.push("keyDisplay");
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				baseArr.push("key");
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				baseArr.push("clefDisplay");
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				baseArr.push("clef");
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				baseArr.push("nvoices");
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				baseArr.push("type");
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				baseArr.push("nbar");
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				baseArr.push("length");
				#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
				{
					#line 9 "F:\\nx3\\src\\nx3\\NPart.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Playground/TestDiff/java/src/nx3/NPart.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class NPart extends haxe.lang.HxObject
{
	public    NPart(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    NPart(haxe.root.Array<nx3.NVoice> voices, nx3.EPartType type, nx3.EClef clef, nx3.EDisplayALN clefDisplay, nx3.EKey key, nx3.EDisplayALN keyDisplay)
	{
		nx3.NPart.__hx_ctor_nx3_NPart(this, voices, type, clef, clefDisplay, key, keyDisplay);
	}
	
	
	public static   void __hx_ctor_nx3_NPart(nx3.NPart __temp_me64, haxe.root.Array<nx3.NVoice> voices, nx3.EPartType type, nx3.EClef clef, nx3.EDisplayALN clefDisplay, nx3.EKey key, nx3.EDisplayALN keyDisplay)
	{
		__temp_me64.nvoices = voices;
		{
			int _g = 0;
			while (( _g < voices.length ))
			{
				nx3.NVoice voice = voices.__get(_g);
				 ++ _g;
				voice.npart = __temp_me64;
			}
			
		}
		
		if (( __temp_me64.nvoices.length > 2 )) 
		{
			throw haxe.lang.HaxeException.wrap("For now, NPart can\'t have more than two voices");
		}
		
		if (( type == null )) 
		{
			__temp_me64.type = nx3.EPartType.Normal;
		}
		 else 
		{
			__temp_me64.type = type;
		}
		
		__temp_me64.clef = clef;
		if (( clefDisplay == null )) 
		{
			__temp_me64.clefDisplay = nx3.EDisplayALN.Layout;
		}
		 else 
		{
			__temp_me64.clefDisplay = clefDisplay;
		}
		
		__temp_me64.key = key;
		if (( keyDisplay == null )) 
		{
			__temp_me64.keyDisplay = nx3.EDisplayALN.Layout;
		}
		 else 
		{
			__temp_me64.keyDisplay = keyDisplay;
		}
		
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.NPart(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.NPart(((haxe.root.Array<nx3.NVoice>) (arr.__get(0)) ), ((nx3.EPartType) (arr.__get(1)) ), ((nx3.EClef) (arr.__get(2)) ), ((nx3.EDisplayALN) (arr.__get(3)) ), ((nx3.EKey) (arr.__get(4)) ), ((nx3.EDisplayALN) (arr.__get(5)) ));
	}
	
	
	public   java.lang.Object iterator()
	{
		return ((java.lang.Object) (new _Array.ArrayIterator<nx3.NVoice>(((haxe.root.Array<nx3.NVoice>) (this.nvoices) ))) );
	}
	
	
	public  int length;
	
	public   int get_length()
	{
		return this.nvoices.length;
	}
	
	
	public  nx3.NBar nbar;
	
	public  nx3.EPartType type;
	
	public  haxe.root.Array<nx3.NVoice> nvoices;
	
	public  nx3.EClef clef;
	
	public  nx3.EDisplayALN clefDisplay;
	
	public  nx3.EKey key;
	
	public  nx3.EDisplayALN keyDisplay;
	
	public   nx3.NVoice getNVoice(int idx)
	{
		if (( ( idx < 0 ) || ( idx > this.nvoices.length ) )) 
		{
			return null;
		}
		 else 
		{
			return this.nvoices.__get(idx);
		}
		
	}
	
	
	@Override public   double __hx_setField_f(java.lang.String field, double value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef420 = true;
			switch (field.hashCode())
			{
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef420 = false;
						this.length = ((int) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef420) 
			{
				return super.__hx_setField_f(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef421 = true;
			switch (field.hashCode())
			{
				case 2046687587:
				{
					if (field.equals("keyDisplay")) 
					{
						__temp_executeDef421 = false;
						this.keyDisplay = ((nx3.EDisplayALN) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef421 = false;
						this.length = ((int) (haxe.lang.Runtime.toInt(value)) );
						return value;
					}
					
					break;
				}
				
				
				case 106079:
				{
					if (field.equals("key")) 
					{
						__temp_executeDef421 = false;
						this.key = ((nx3.EKey) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3374309:
				{
					if (field.equals("nbar")) 
					{
						__temp_executeDef421 = false;
						this.nbar = ((nx3.NBar) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 1291990680:
				{
					if (field.equals("clefDisplay")) 
					{
						__temp_executeDef421 = false;
						this.clefDisplay = ((nx3.EDisplayALN) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3575610:
				{
					if (field.equals("type")) 
					{
						__temp_executeDef421 = false;
						this.type = ((nx3.EPartType) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3056330:
				{
					if (field.equals("clef")) 
					{
						__temp_executeDef421 = false;
						this.clef = ((nx3.EClef) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -1969833105:
				{
					if (field.equals("nvoices")) 
					{
						__temp_executeDef421 = false;
						this.nvoices = ((haxe.root.Array<nx3.NVoice>) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef421) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef422 = true;
			switch (field.hashCode())
			{
				case 633088090:
				{
					if (field.equals("getNVoice")) 
					{
						__temp_executeDef422 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getNVoice"))) );
					}
					
					break;
				}
				
				
				case 1182533742:
				{
					if (field.equals("iterator")) 
					{
						__temp_executeDef422 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("iterator"))) );
					}
					
					break;
				}
				
				
				case 2046687587:
				{
					if (field.equals("keyDisplay")) 
					{
						__temp_executeDef422 = false;
						return this.keyDisplay;
					}
					
					break;
				}
				
				
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef422 = false;
						if (handleProperties) 
						{
							return this.get_length();
						}
						 else 
						{
							return this.length;
						}
						
					}
					
					break;
				}
				
				
				case 106079:
				{
					if (field.equals("key")) 
					{
						__temp_executeDef422 = false;
						return this.key;
					}
					
					break;
				}
				
				
				case 974314479:
				{
					if (field.equals("get_length")) 
					{
						__temp_executeDef422 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("get_length"))) );
					}
					
					break;
				}
				
				
				case 1291990680:
				{
					if (field.equals("clefDisplay")) 
					{
						__temp_executeDef422 = false;
						return this.clefDisplay;
					}
					
					break;
				}
				
				
				case 3374309:
				{
					if (field.equals("nbar")) 
					{
						__temp_executeDef422 = false;
						return this.nbar;
					}
					
					break;
				}
				
				
				case 3056330:
				{
					if (field.equals("clef")) 
					{
						__temp_executeDef422 = false;
						return this.clef;
					}
					
					break;
				}
				
				
				case 3575610:
				{
					if (field.equals("type")) 
					{
						__temp_executeDef422 = false;
						return this.type;
					}
					
					break;
				}
				
				
				case -1969833105:
				{
					if (field.equals("nvoices")) 
					{
						__temp_executeDef422 = false;
						return this.nvoices;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef422) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   double __hx_getField_f(java.lang.String field, boolean throwErrors, boolean handleProperties)
	{
		{
			boolean __temp_executeDef423 = true;
			switch (field.hashCode())
			{
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef423 = false;
						if (handleProperties) 
						{
							return ((double) (this.get_length()) );
						}
						 else 
						{
							return ((double) (this.length) );
						}
						
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef423) 
			{
				return super.__hx_getField_f(field, throwErrors, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef424 = true;
			switch (field.hashCode())
			{
				case 633088090:
				{
					if (field.equals("getNVoice")) 
					{
						__temp_executeDef424 = false;
						return this.getNVoice(((int) (haxe.lang.Runtime.toInt(dynargs.__get(0))) ));
					}
					
					break;
				}
				
				
				case 1182533742:
				{
					if (field.equals("iterator")) 
					{
						__temp_executeDef424 = false;
						return this.iterator();
					}
					
					break;
				}
				
				
				case 974314479:
				{
					if (field.equals("get_length")) 
					{
						__temp_executeDef424 = false;
						return this.get_length();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef424) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("keyDisplay");
		baseArr.push("key");
		baseArr.push("clefDisplay");
		baseArr.push("clef");
		baseArr.push("nvoices");
		baseArr.push("type");
		baseArr.push("nbar");
		baseArr.push("length");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


